docs/administrators/policy_cookbook.md

```diff
@@ -798,3 +798,152 @@ The implementation can be modified to apply only to specific resources, collecti
+    "checksum": "",
+    "collection_id": 10010,
+    "condition_input": [
+      {
+        "key": "resc_hier",
+        "value": "demoResc"
+      },
+      {
+        "key": "selected_hierarchy",
+        "value": "demoResc"
+      }
+    ],
+    "data_access": "",
+    "data_access_index": 0,
+    "data_comments": "",
+    "data_create": "01740005951",
+    "data_expiry": "00000000000",
+    "data_id": 10015,
+    "data_map_id": 0,
+    "data_mode": "384",
+    "data_modify": "01740006513",
+    "data_owner_name": "rods",
+    "data_owner_zone": "tempZone",
+    "data_size": 12,
+    "data_type": "generic",
+    "destination_resource_name": "",
+    "file_path": "/var/lib/irods/Vault/home/rods/foo",
+    "flags": 0,
+    "in_pdmo": "",
+    "is_replica_current": true,
+    "next": null,
+    "object_path": "/tempZone/home/rods/foo",
+    "other_flags": 0,
+    "registering_user_id": 0,
+    "replica_number": 0,
+    "replica_status": 1,
+    "resource_hierarchy": "demoResc",
+    "resource_id": 10013,
+    "resource_name": "demoResc",
+    "special_collection": null,
+    "status_string": "",
+    "sub_path": "",
+    "version": "",
+    "write_flag": 0
+  },
+  "data_object_input_replica_flag": 1,
+  "data_object_input": {
+    "condition_input": [
+      {
+        "key": "resc_hier",
+        "value": "demoResc"
+      },
+      {
+        "key": "selected_hierarchy",
+        "value": "demoResc"
+      }
+    ],
+    "data_size": -1,
+    "in_pdmo": "",
+    "in_use": true,
+    "l3descInx": 3,
+    "lock_file_descriptor": 0,
+    "number_of_threads": 0,
+    "object_path": "/tempZone/home/rods/foo",
+    "offset": 0,
+    "open_flags": 0,
+    "open_type": 2,
+    "operation_status": 0,
+    "operation_type": 0,
+    "other_data_object_info": null,
+    "plugin_data": null,
+    "purge_cache_flag": 0,
+    "remote_l1_descriptor_index": 0,
+    "remote_zone_host": null,
+    "replica_status": 1,
+    "replica_token": "",
+    "replication_data_object_info": null,
+    "source_l1_descriptor_index": 0,
+    "special_collection": null
+  },
+  "stage_flag": 0
+}
+```
+
+In order to access this information via the microservice, we need a handle to a JSON structure and a JSON pointer indicating which key's value we want. For example, if we wanted to know the creation time of the replica, we would use `"/data_object_info/data_create"`.
+
+In this example, we will implement a dynamic PEP for the DataObjRead API, which uses an `OpenedDataObjInp` structure as part of its signature. In this PEP, we do not know the logical path of the data object which is being read and the `OpenedDataObjInp` structure does not have this information. However, it does give us access to an L1 file descriptor. Using this, we can access information about the opened replica which already resides in the iRODS agent's memory. Here is the implementation:
+```python
+pep_api_data_obj_read_pre(*instance_name, *comm, *opened_data_obj_inp, *data_obj_read_out_bbuf) {
```

Comment:
Move the opening curly brace to its own line so it matches the rest of the cookbook examples.